### PR TITLE
Fixed o3de Release compilation 

### PR DESCRIPTION
**Pull request by vmednonogov**
## What does this PR do?

Fixed o3de Release compilation 

## How was this PR tested?

It is verifying via Jenkins build now (iOS, Android, Linux)
http://10.0.1.100:8080/job/MadWorld/job/Client-O3DE/job/build%252Fvmedn%252Ftest-MAD-15638-fixed-Release-compilation-o3de/

Also verified PC build and iOS build locally

### CODE REVIEW

File: Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ namespace AZ
                     m_statusAfterExecution = buffer.status;
                     if (m_statusAfterExecution == MTLCommandBufferStatusError)
                     {
-                        const char * cbLabel = [ buffer.label UTF8String ];
+                        [[maybe_unused]] const char * cbLabel = [ buffer.label UTF8String ];
```
Comment (alexeykostin) [inline Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp:59]: As soon as you are touching this code, please add `\n` to `AZ_Printf`'s in the new code below

**Review by alexeykostin — APPROVED** [review]

File: Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp (hunk 1 above)
Comment (vmednonogov) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp:59]: Updated

Comment (aefimov3) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp:59]: Here is the real log piece from a test:
<20:36:05> [Error] (RHI) - Command Buffer MergedGroupCB_1 failed to execute
<20:36:05> (RHI) - Command set Root, state completed
<20:36:05> (RHI) - Command set Shadows.Cascades.0, state failed
<20:36:05> (RHI) - Execution of this command generated an unserviceable GPU page fault. This error maybe caused by buffer read/write attribute mismatch or out of boundary access.
As you see there are proper new lines.

Comment (aefimov3) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp:59]: I believe we do not need \n.

Comment (aefimov3) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp:59]: There is an extra new line after the last message, so I belie we need no \n here on iOS. Please remove.

Comment (vmednonogov) [reply Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueueCommandBuffer.cpp:59]: Reverted

**Review by Twolewis — APPROVED** [review]

**Review by aefimov3 — APPROVED** [review]

**Comment by Twolewis** [comment]
I'm merging this preemptively since I'm blocked by it ;)  thanks!!

**Pull request merged**